lib/dry/system/component.rb

```diff
@@ -1,49 +1,95 @@
-require 'inflecto'
+require 'concurrent/map'
+
+require 'dry-equalizer'
+require 'dry/system/loader'
 
 module Dry
   module System
     class Component
-      attr_reader :loader
-      attr_reader :identifier, :path, :file
+      include Dry::Equalizer(:identifier, :path)
+
+      PATH_SEPARATOR = '/'.freeze
+      WORD_REGEX = /\w+/.freeze
+
+      attr_reader :identifier, :path, :file, :options, :loader
+
+      def self.new(*args)
+        cache.fetch_or_store(args.hash) do
+          name, options = args
+          ns, sep = options.values_at(:namespace, :separator).map(&:to_s)
```

Comment:
If we _did_ want to make component initialization more robust, I'd recommend a default `"."` value for the separator, and allowing it to be created without an options hash.